design/proof-network.md

```diff
@@ -0,0 +1,257 @@
+- validators: they keep track of submitted and missed proofs, and trigger
+  slashing on-chain
+
+The validators form a consensus network that allows them to agree on blocks of
+transactions. Storage providers submit transactions containing a storage proof
+to one of the validators, which includes it in a block of transactions.
+Validators also monitor the on-chain marketplace to check when storage proofs
+are missed.
+
+                                                 consensus network
+
+                                       --------------------------------------
+     storage providers                 |                                    |
+                                       |               validator            |
+    ---------------------              |                ^    ^              |
+    |                   |              |               /      \             |
+    |    provider       |              |              /        \            |
+    |                                                v          v           |
+    |      provider  <--------------------->  validator <----> validator    |
+    |                                                ^          ^           |
+    |  provider         |              |              \        /            |
+    |                   |              |               \      /             |
+    ---------------------              |                v    v              |
+           ^                           |               validator            |
+           |                           |                                    |
+           |                           --------------------------------------
+           |                                          ^
+           |                                          |
+           |             ethereum                     |
+           |                                          |
+           |        ------------------                |
+           |        |                |                |
+           \------- |  marketplace   | <--------------/
+                    |                |
+                    ------------------
+
+Transactions
+------------
+
+The types of transactions that can be included in blocks are:
+
+- `StorageProof(slot id, period, inputs, proof)`
+- `MissingProof(slot id, period, inputs)`
+
+Each transaction is signed by its sender. A *slot id* parameters refers to a
+[slot in a storage request][4] on the marketplace. It uniquely identifies the
+data for which a storage proof is required. The *period* refers to a [time
+interval][5] in which the storage proof should be submitted. By *proof* we mean
+a [zero-knowledge proof][5], and by *inputs* we mean its public inputs.
+
+#### StorageProof ####
+
+A storage provider sends a `StorageProof` transaction to a validator to indicate
+to the network that it calculated a storage proof as required by the
+marketplace. This validator includes it in its next block.
```

Comment:
The proof is sent to only one validator or the entire network of validators? This validator includes it in its next block? Or rather the validator in charge of producing the next block adds the transaction to it?

Reply:
The proof is sent to only one validator. In the Mysticeti design, all validators produce a block each round, so the proof is included in the next block of the validator that you sent it to.

If it fails to provide you with proof that it included the transaction, then you're free to go to another validator.

Reply:
How are proofs/transactions propagated across validators?

Reply:
> How are proofs/transactions propagated across validators?

As part of the consensus protocol blocks of transactions are exchanged between validators.